sparse_ho/criterion/cross_val.py

```diff
@@ -49,14 +49,18 @@ def _initialize(self, model, X):
             self.dict_crits[i].idx_val = idx_val
             self.dict_models[i] = copy.deepcopy(model)
 
-    def get_val(self, model, log_alpha, tol=1e-3):
-        val = 0
+    def get_val(self, model, X, y, log_alpha, tol=1e-3, monitor=None):
+        if self.dict_crits is None:
+            self._initialize(model, X)
+        value_outer = 0
         for i in range(self.n_splits):
-            vali = self.dict_crits[i].get_val(self.models[i], log_alpha,
-                                              tol=tol)
-            val += vali
-        val /= self.n_splits
-        return val
+            vali = self.dict_crits[i].get_val(
```

Comment:
@qb3 what do you think of `value_outer = np.mean([self.dict_crits[i].... for i in range(self.n_splits)]` ? The line is long but you avoid setting value_outer to 0, then the division step

Reply:
Done, thx :)